src/Microsoft.ML.Transforms/NAReplaceTransform.cs

```diff
@@ -178,89 +218,51 @@ private static string TestType<T>(ColumnType type)
 
         // Marks if the replacement values in given slots of _repValues are the default value.
         // REVIEW: Currently these arrays are constructed on load but could be changed to being constructed lazily.
-        private BitArray[] _repIsDefault;
-
-        // The isNA delegates, parallel to Infos.
-        private readonly Delegate[] _isNAs;
+        private readonly BitArray[] _repIsDefault;
 
-        public override bool CanSaveOnnx => true;
-
-        /// <summary>
-        /// Convenience constructor for public facing API.
-        /// </summary>
-        /// <param name="env">Host Environment.</param>
-        /// <param name="input">Input <see cref="IDataView"/>. This is the output from previous transform or loader.</param>
-        /// <param name="name">Name of the output column.</param>
-        /// <param name="source">Name of the column to be transformed. If this is null '<paramref name="name"/>' will be used.</param>
-        /// <param name="replacementKind">The replacement method to utilize.</param>
-        public NAReplaceTransform(IHostEnvironment env, IDataView input, string name, string source = null, ReplacementKind replacementKind = ReplacementKind.DefaultValue)
-            : this(env, new Arguments() { Column = new[] { new Column() { Source = source ?? name, Name = name } }, ReplacementKind = replacementKind }, input)
+        protected override void CheckInputColumn(ISchema inputSchema, int col, int srcCol)
         {
+            var type = inputSchema.GetColumnType(srcCol);
+            string reason = TestType(type);
+            if (reason != null)
+                //IVAN: not sure about schema mismatch
```

Comment:
>t sure about schema mismatch [](start = 26, length = 28)

no, it looks right

Reply:
Look on TestType and what kind of string it returns.
No way I can convert it to current SchemaMismatch wording.

---
In reply to: [217835064](https://github.com/dotnet/machinelearning/pull/917#discussion_r217835064) [](ancestors = 217835064)

Reply:
oh yuck. Ok, then do `ExceptParam`

---
In reply to: [217836776](https://github.com/dotnet/machinelearning/pull/917#discussion_r217836776) [](ancestors = 217836776,217835064)